apps/wallet/src/background/accounts/zk/ZkAccount.ts

```diff
@@ -251,4 +243,69 @@ export class ZkAccount
 		signatureBytes.set(bytes, 1);
 		return toB64(signatureBytes);
 	}
+
+	#areCredentialsValid(
+		currentEpoch: number,
+		activeNetwork: NetworkEnvType,
+		credentials?: CredentialData,
+	): credentials is CredentialData {
+		if (!credentials) {
+			return false;
+		}
+		const { maxEpoch, network } = credentials;
+		return (
+			activeNetwork.env === network.env &&
+			activeNetwork.customRpcUrl === network.customRpcUrl &&
+			currentEpoch <= maxEpoch
+		);
+	}
+
+	async #doLogin() {
+		const { provider, claims } = await this.getStoredData();
+		const { email, sub, aud, iss } = await deobfuscate<JwtSerializedClaims>(claims);
+		const epoch = await getCurrentEpoch();
+		const { ephemeralKeyPair, nonce, randomness, maxEpoch } = prepareZKLogin(Number(epoch));
+		const jwt = await zkLogin({ provider, nonce, loginHint: sub });
+		const decodedJWT = decodeJwt(jwt);
+		if (
+			decodedJWT.aud !== aud ||
+			decodedJWT.email !== email ||
```

Comment:
Iirc the JWT might not have the email as part of its fields, no? @mskd12 can you confirm?

Reply:
Yeah, I think for google this should be rare but for other providers might be common. I think I will remove it the other checks should be enough

Reply:
@pchrysochoidis I thought we were getting the email as a "display string", no? We had a disc at some point to find better "display strings". It might be different for each provider, e.g., Google -> email, Twitch -> username.

But to Kostas's point, I think the bigger problem is that there might be edge cases where any claim we pick (email or username) might not be there.

Reply:
Yeah the only usage now is as a nickname of the account, that's why I removed it from the check